tla-io/src/main/scala/at/forsyte/apalache/tla/imp/SanyImporter.scala

```diff
@@ -106,34 +106,41 @@ class SanyImporter(sourceStore: SourceStore, annotationStore: AnnotationStore) e
    * into it, in order to call SANY.
    *
    * @param source
-   *   the text source
+   *   the text source for the root module
+   * @param aux
+   *   the text sources for any auxiliary modules
    * @return
    *   the pair (the root module name, a map of modules)
    */
-  def loadFromSource(source: Source): (String, Map[String, TlaModule]) = {
+  def loadFromSource(source: Source, aux: Seq[Source] = Seq()): (String, Map[String, TlaModule]) = {
+    val tempDir = Files.createTempDirectory("sanyimp").toFile
+    try {
+      val rootModule = saveTlaFile(tempDir, source)
+      aux.foreach(saveTlaFile(tempDir, _))
+      loadFromFile(rootModule)
+    } finally {
+      tempDir.delete()
+    }
+  }
+
+  private def saveTlaFile(dir: File, source: Source): File = {
     val moduleName = moduleNameOfSource(source) match {
       case Some(n) => n
-      case None    => throw new SanyImporterException(s"No module name found in given source")
+      case None    => throw new SanyImporterException(s"No module name found in source for module")
     }
 
-    val tempDir = Files.createTempDirectory("sanyimp").toFile
-    val temp = new File(tempDir, moduleName + ".tla")
+    val temp = new File(dir, moduleName + ".tla")
+    // write the contents to a temporary file
+    val pw = new PrintWriter(temp)
     try {
-      // write the contents to a temporary file
-      val pw = new PrintWriter(temp)
-      try {
-        Try(source.getLines()) match {
-          case Success(lines) => lines.foreach(line => pw.println(line))
-          case Failure(e)     => throw new FileNotFoundException("Source not found: " + e.getMessage)
-        }
-      } finally {
-        pw.close()
+      Try(source.getLines()) match {
+        case Success(lines) => lines.foreach(line => pw.println(line))
+        case Failure(e)     => throw new FileNotFoundException("Source not found: " + e.getMessage)
```

Comment:
Looks like this will never happen (as we only wrap `String`s into `Source`s) and could be avoided if we just directly passed a `String`.

Reply:
This code wasn't added by PR, just FYI, it was only moved.

We are currently only converting source's to strings, but if this was used to stream data from stdin, for example, then this error could arise, no?

Reply:
I moved this so we have this safety check when we first call getLines now, and we jus use try's everywhere, so i think it's a bit cleaner?

fe66642038af698a4809607988860bb653d40f8d